src/Generator/Resolver/UniqueValuesPool.php

```diff
@@ -95,4 +102,43 @@ public function add(UniqueValue $value)
 
         $this->pool[$valueId][] = $value->getValue();
     }
+
+    private function objectsAreIdentical($o1, $o2): bool
```

Comment:
I would move this in a util class static method that could be unit tested separately

Reply:
Agreed, was just a quick fix to test something out. Care to suggest a name / location?

Also,  (and besides a some unit tests) this would need testing in more real-world applications, to get a view on the performance impact. And to make sure that no unintended behaviour changes occured.

Anyway: this allows me to use `(unique)` on m:n relations which may be cyclic, while not running into an infinite loop.

Any thoughts on the basic logic?